File: Solutions/6kyu/6kyu_numerical_palindrome_3.py
from itertools import combinations_with_replacement as cwr

def palindrome(num):
    if not isinstance(num, int) or num < 0: return "Not valid"
    
    num = str(num)
    count = 0
    
    # generate indexes for all possible list slices
    for i,j in cwr(range(len(num)+1),2):
        check = num[i:j]
        if check == check[::-1] and len(check)>1: count += 1
    
    return count

File: Solutions/6kyu/6kyu_numerical_palindrome_3.js
function palindrome(num) { 
  if (typeof num != 'number' || num < 0 || num % 1 != 0) { return 'Not valid' }
  var num = String(num)
  var count = 0
  for (var i of genIndexes(num.length)) {
    var check = num.substring(i[0],i[1])
    if (check == check.split('').reverse().join('') && check.length > 1) { count += 1 }  
  }
  return count
}

// generate indexes for all possible string slices
function* genIndexes(lim){
  for (var i=0; i <= lim; i++){
    for (var j=i; j <= lim; j++){
      yield [i,j]
    }
  }
}
